### PR TITLE
Adding a few annotators to external

### PR DESCRIPTION
**Pull request by danyaljj**

### CODE REVIEW

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java [hunk 1]
```diff
@@ -392,6 +392,25 @@ private View createRestrictedView(ITransformer<View, Double> scoreTransformer) {
...
+     * Given char-begin-offset and char-end-offset, it returns the constituents with some overlap with the input char offsets.
+     * @param charStart the begin char index
+     * @param charEnd the end char index (one-past-the-end indexing)
+     */
+    public List<Constituent> getConstituentsOverlappingCharSpan(int charStart, int charEnd) {
+
+        Set<Constituent> output = new HashSet<>();
+        for(Constituent c : this.constituents) {
+            if((c.startCharOffset <= charStart && c.endCharOffset >= charStart) ||
+                    (c.startCharOffset >= charEnd && c.endCharOffset <= charEnd) ||
```
Comment (danyaljj) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java]: Good catch!

**Comment by danyaljj** [comment]
Addressed @bhargav 's comment. Waiting for @mssammon 's golden review.

File: external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordOpenIEHandler.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+public class StanfordOpenIEHandler extends Annotator {
+
+    public static final String viewName = "STANFORD_OPENIE";
+
+    StanfordCoreNLP pipeline = null;
+
+    public StanfordOpenIEHandler() {
+        super(viewName, new String[]{}, true);
+    }
+
```
Comment (mssammon) [inline external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordOpenIEHandler.java]: should we also provide a constructor with a resourcemanager to override default options?

File: external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordRelationsHandler.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    private static Constituent createConstituentGivenMention(EntityMention rm, TextAnnotation ta) {
+        List<CoreLabel> tokens = rm.getSentence().get(CoreAnnotations.TokensAnnotation.class);
+        int extentCharStart = tokens.get(rm.getExtentTokenStart()).beginPosition();
+        int extentCharEnd = tokens.get(rm.getExtentTokenEnd()).endPosition();
+        int headCharStart = tokens.get(rm.getHeadTokenStart()).beginPosition();
+        int headCharEnd = tokens.get(rm.getHeadTokenEnd()).endPosition();
+
+        List<Constituent> extentCons = ta.getView(ViewNames.TOKENS).getConstituentsOverlappingCharSpan(extentCharStart, extentCharEnd);
```
Comment (mssammon) [inline external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordRelationsHandler.java]: it looks like this is doing some heuristic token boundary fixing.  Please add a brief javadoc that gives some idea of behavior.

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java [hunk 4]
```diff
@@ -392,6 +392,25 @@ private View createRestrictedView(ITransformer<View, Double> scoreTransformer) {
         return list;
     }
 
+    /**
+     * Given char-begin-offset and char-end-offset, it returns the constituents with some overlap with the input char offsets.
+     * @param charStart the begin char index
+     * @param charEnd the end char index (one-past-the-end indexing)
+     */
+    public List<Constituent> getConstituentsOverlappingCharSpan(int charStart, int charEnd) {
+        assert charStart <= charEnd: "The start offset should be smaller than the end offset";
```
Comment (bhargav) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java]: When an assertion fails, an AssertionError is thrown with the message statement. So I think it is similar to throwing an Exception instance (like RuntimeException). I don't know if there are any limitations of using assertions.

**Comment by bhargav** [comment]
One other comment was that we should try to have a common place to put all attributes to Relation, Mention, Coreference annotations.

For ACE and ERE, we try to use the same attribute names. We can do this as a separate PR/issue.

File: external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordCorefHandler.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    @Override
+    public void initialize(ResourceManager rm) {
+        Properties props = new Properties();
+        props.setProperty("annotators", "tokenize,ssplit,pos,lemma,ner,parse,mention,coref");
+        this.pipeline = new StanfordCoreNLP(props);
+    }
+
+    @Override
+    protected void addView(TextAnnotation ta) throws AnnotatorException {
+        Annotation document = new Annotation(ta.text);
```
Comment (danyaljj) [inline external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordCorefHandler.java]: Good point! Will create an issue.

File: external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordOpenIEHandler.java (hunk 2 above)
Comment (danyaljj) [inline external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordOpenIEHandler.java]: True. I'm not sure what options are worth changing. We can do that case by case.

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java (hunk 4 above)
Comment (danyaljj) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java]: 👍

File: external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordOpenIEHandler.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import edu.stanford.nlp.pipeline.Annotation;
+import edu.stanford.nlp.pipeline.StanfordCoreNLP;
+import edu.stanford.nlp.util.CoreMap;
+import edu.stanford.nlp.util.Pair;
+
+import java.util.*;
+
+/**
+ * Do openIE with Stanford CoreNLP
+ */
```
Comment (danyaljj) [inline external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordOpenIEHandler.java]: done.

File: external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordRelationsHandler.java (hunk 3 above)
Comment (danyaljj) [inline external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordRelationsHandler.java]: done.

**Comment by danyaljj** [comment]
I think I addressed all the comments. Going to merge this soon.

**Pull request merged**